addon/authenticators/oidc.js

```diff
@@ -192,7 +192,9 @@ export default class OidcAuthenticator extends BaseAuthenticator {
 
       // Failed refresh
       const isBadRequest = isBadRequestResponse(response);
-      if (isBadRequest) return Promise.reject(data);
+      if (isBadRequest) {
+        location.replace(getAbsoluteUrl(this.config.afterLogoutUri || ""));
+      }
```

Comment:
@luytena @kaldras Added this to intercept the auth refresh in case the token refresh fails with a `400`. This will prevent any further requests with a invalid token. Not sure if we should save the current URL as `data.nextURL` as well, like in the `handleUnauthorized` implementation.

Without this interception, the token refresh request fails, but no action is taken. The next request with the old invalid token will fail and trigger the `handleUnauthorized` handler.

Reply:
By doing this, we circumvent the authentication and lifecycle handling done by ember-simple-auth. It also prevents anybody using the addon from making use of the handlers. I would therefore avoid immediately redirecting upon a failed token refresh, without letting ember-simple-auth handle it accordingly.

Reply:
Good point! But I think we still have to act upon the first invalid request. --> amended change.